File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] pos = {0, 0};
        int maxX = (board[0] - 1) / 2;
        int maxY = (board[1] - 1) / 2;

        for (String k : keyinput) {
            if (k.equals("up") && pos[1] < maxY) {
                pos[1]++;
            } else if (k.equals("down") && pos[1] > -maxY) {
                pos[1]--;
            } else if (k.equals("right") && pos[0] < maxX) {
                pos[0]++;
            } else if (k.equals("left") && pos[0] > -maxX) {
                pos[0]--;
            }
        }

        return pos;
    }
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    const pos = [0, 0];
    const maxX = (board[0] - 1) / 2;
    const maxY = (board[1] - 1) / 2;

    for (const k of keyinput) {
        if (k === "up" && pos[1] < maxY) {
            pos[1]++;
        } else if (k === "down" && pos[1] > -maxY) {
            pos[1]--;
        } else if (k === "right" && pos[0] < maxX) {
            pos[0]++;
        } else if (k === "left" && pos[0] > -maxX) {
            pos[0]--;
        }
    }

    return pos;
}